src/shell/Shell.cpp

```diff
@@ -1018,6 +1021,40 @@ static void parseArguments(int argc, char* argv[], ArgParser& argParser)
             jitVerbose = 1;
         } else if (*it == "--jit-verbose-color") {
             jitVerbose = 2;
+        } else if (*it == "--mapdirs") {
+            it++;
+            while (it != args.end() && nonstd::to_string(*it).find("--") == std::string::npos) {
+                if (it + 1 == args.end()) {
+                    break;
+                }
+
+                uvwasi_preopen_s preopen;
+                preopen.real_path = reinterpret_cast<char*>(calloc(1, it->length() + 1));
+                it->nonstd::sv_lite::string_view::copy(const_cast<char*>(preopen.real_path), it->length());
+
+                it++;
+
+                preopen.mapped_path = reinterpret_cast<char*>(calloc(1, it->length() + 1));
+                it->nonstd::sv_lite::string_view::copy(const_cast<char*>(preopen.mapped_path), it->length());
```

Comment:
Will these strings allocated by `calloc` be correctly freed at the end of Walrus?

Reply:
No, I forgot about those. After running valgrind I fixed these and also added initialization to wasi envrionment variables, so valgrind did not return any errors.